gallery_dl/extractor/motherless.py

```diff
@@ -0,0 +1,308 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://motherless.com/"""
+
+from .common import Extractor, Message
+from .. import text
+import re
+from datetime import datetime, timedelta, timezone
+from html import unescape
+
+ROOT_URL_PATTERN = r"(?:https?://)?motherless\.com"
+
+
+class MotherlessExtractor(Extractor):
+    """Base class for motherless extractors"""
+
+    category = "motherless"
+    root = "https://motherless.com"
+    filename_fmt = "{id} {title}.{extension}"
+
+
+class MotherlessMediaExtractor(MotherlessExtractor):
+    """Extractor for a single image/video from motherless.com"""
+
+    pattern = ROOT_URL_PATTERN + "/((?!GV|GI|G)[A-Z0-9]+)$"
+    example = "https://motherless.com/ABC123"
+    directory_fmt = ("{category}",)
+
+    def items(self):
+        image_url, data = self.get_image()
+        yield Message.Directory, data
+        yield Message.Url, image_url, data
+
+    def get_image(self):
+        self.page_data = self.request(self.url).text
+
+        try:
+            # Find image source url.
+            self.subcategory = "image"
+
+            image_url_search = re.search(f'<link rel="image_src" type="image/([a-z]+)" href="(.+)">', self.page_data)
+            extension = image_url_search.group(1)
+            media_url = image_url_search.group(2)
+            id = get_image_id(media_url)
+
+        except AttributeError:
+            # No image, find video url.
+            self.subcategory = "video"
+
+            video_url_search = re.search("__fileurl = '(.+)'", self.page_data)
+            extension = "mp4"
+            media_url = video_url_search.group(1)
+            id = get_video_id(media_url)
+
+        data = {
+            "url": self.url,
+            "title": get_media_title(self.page_data),
+            "id": id,
+            "filename": id,
+            "extension": extension,
+            "date": get_media_date(self.page_data),
+            "uploader": get_media_uploader(self.page_data),
+            "tags": get_media_tags(self.page_data)}
+
+        return media_url, data
+
+
+class MotherlessMediaInGalleryExtractor(MotherlessMediaExtractor):
+    """Extractor for a single image/video from a gallery from motherless.com"""
+
+    directory_fmt = ("{category}", "{gallery_id} {gallery_title}")
+    pattern = ROOT_URL_PATTERN + "/(?:GI?)([A-Z0-9]+)/([A-Z0-9]+)"
```

Comment:
If the ID includes the prefix, then `get_images` and `get_videos` must do further checks to construct the URL for the only-images pages and the only-videos pages. They cannot simply append the ID to the root URL. 

This also would result in images, videos, or both, from the same gallery going into three different directories when downloaded at different instances due to different IDs for the same gallery - unless we do even more checks to avoid this! Checking for the type of gallery given at the first hurdle simplifies things later on. 

See here, from `get_images`:
```
page = extractor.request(f"{extractor.root}/{extractor.gallery_id}?page={n}").text
```

`extractor.gallery_id` could start with either `GI` or `G`. Without the prefix included there is no check. Similarly for `get_videos`. Compare this to what we currently have working:

```
page = extractor.request(f"{extractor.root}/GI{extractor.gallery_id}?page={n}").text
```